### DISCUSSION TITLE
usqInfiniteQuery hook query key not being set

### DISCUSSION THREAD

**Discussion by cubecleveland** [TanStack/query#3467] [Q&A]
Helo, im using the Infinite query hook to fetch scrolling info from a backend..

My code look like this

useInfiniteQuery(
    [
      id,
      "search",
      queryFilter
    ],
    fetch,
    {
      getNextPageParam: (lastPage, pages) => {
        return pages.length < lastPage.last_page
          ? lastPage.last_page
          : undefined;
      },
    }
  );

My hope is that queryFilter will in term alwasy cache different queries based on the value, but it seems like the query keys bounce and the queries are being overwritten .

the same exact method works flawlessly when using useQuery to fetch regular queries.....
Are we missing something, is there any other way to ensure the perisstnet of the query keys using useInifinteQuery ?

**Comment by TkDodo** [comment]
with regards to the query key, there is no difference between `useQuery` and `useInfiniteQuery`. If your key contains some filters, you will get a separate cache entry per filter. The devtools should show that.

Please provide a codesandbox reproduction if that's not the case.
